src/main/java/org/optaplanner/springboottaskassigning/solver/DefaultSolverManager.java

```diff
@@ -70,83 +87,83 @@ public synchronized void solve(Object problemId, Solution_ planningProblem,
         CompletableFuture<Solution_> solverFuture = CompletableFuture.supplyAsync(newSolverTask::startSolving, solverExecutorService);
         solverFuture.handle((solution_, throwable) -> {
             if (throwable != null) {
-                throw new RuntimeException("Error in handling CompletableFuture of problem (" + problemId + ").", throwable.getCause());
+                logger.error("Exception while solving problem (" + problemId + ").", throwable.getCause());
+                if (onException != null) {
+                    eventHandlerExecutorService.submit(() -> onException.accept(throwable.getCause()));
+                }
             }
             return eventHandlerExecutorService.submit(() -> onSolvingEnded.accept(solution_));
         });
-        problemIdToSolverTaskMap.put(problemId, newSolverTask);
-        logger.info("A new solver task was created with problemId ({}).", problemId);
     }
 
     @Override
     public void stopSolver(Object problemId) {
+        logger.debug("Stopping solver of problemId ({}).", problemId);
         if (!problemIdToSolverTaskMap.containsKey(problemId)) {
-            throw new IllegalArgumentException("Solver (" + problemId + ") does not exist.");
+            throw new IllegalArgumentException("Problem (" + problemId + ") was not submitted.");
         }
-        stopSolverTask(problemIdToSolverTaskMap.get(problemId));
+        problemIdToSolverTaskMap.get(problemId).stopSolver();
+    }
+
+    @Override
+    public boolean problemSubmitted(Object problemId) {
+        return problemIdToSolverTaskMap.containsKey(problemId);
     }
 
     @Override
     public Solution_ getBestSolution(Object problemId) {
         logger.debug("Getting best solution of problemId ({}).", problemId);
+        if (!problemIdToSolverTaskMap.containsKey(problemId)) {
+            logger.error("Problem (" + problemId + ") was not submitted.");
+        }
         SolverTask<Solution_> solverTask = problemIdToSolverTaskMap.get(problemId);
         return solverTask == null ? null : solverTask.getBestSolution();
```

Comment:
I have to offer the `Optional` again; I know it was discussed while by battery died :) yet this case in my opinion screams for `Optional`.

Reply:
I see your point but it's the same when we decided not to use `Optional`, let's discuss it again in the next sync up.